src/Quartz/Job/SendMailJob.cs

```diff
@@ -80,6 +82,7 @@ public virtual Task Execute(IJobExecutionContext context)
         {
             JobDataMap data = context.MergedJobDataMap;
 
+#if MAIL
```

Comment:
We can hide the entire class behind the pre-processor directive. Users can be guided to have a look at the implementation and and write own Core version with mail library they like the most.